### PR TITLE
Switch from detecting faas.id to faas.instance in the gcp detector.

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Context: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/679

faas.id was removed from the semantic conventions: https://github.com/open-telemetry/opentelemetry-specification/pull/3188

We were previously using it improperly to store the instance id of the faas, which should be mapped to faas.instance instead.

**Testing:**

Unit tests

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
I believe since this is an end-user impacting breaking change it should be done via feature-gates https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#end-user-impacting-changes

**Comment by dashpole** [comment]
@codeboten I added a feature gate, and changed the PR from `breaking` to `deprecation`.

**Comment by dashpole** [comment]
@codeboten @dmitryax @Aneurysm9 PTAL if you have time.

File: processor/resourcedetectionprocessor/internal/gcp/gcp.go [hunk 1]
```diff
@@ -23,6 +24,12 @@ const (
 	TypeStr = "gcp"
 )
 
+var removeGCPFaasID = featuregate.GlobalRegistry().MustRegister(
+	"processor.resourcedetection.removeGCPFaasID",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("Remove faas.id from the GCP detector. Use faas.instance instead."),
+	featuregate.WithRegisterFromVersion("v0.85.0"))
```
Comment (codeboten) [inline processor/resourcedetectionprocessor/internal/gcp/gcp.go:31]:
```suggestion
	featuregate.WithRegisterFromVersion("v0.87.0"))
```

**Review by codeboten — APPROVED** [review]
Thanks @dashpole, just one suggestion

**Comment by codeboten** [comment]
Please resolve the conflict

File: processor/resourcedetectionprocessor/internal/gcp/gcp.go (hunk 1 above)
Comment (codeboten) [reply processor/resourcedetectionprocessor/internal/gcp/gcp.go:31]: @dashpole not seeing this change... maybe it's something weird in github

**Comment by dashpole** [comment]
@codeboten sorry.  Forgot to pull in origin when I fixed go.mod

**Review by codeboten — APPROVED** [review]
np! thanks for following up

**Pull request merged**